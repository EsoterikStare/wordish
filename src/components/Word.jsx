import React from "react";
import { createUseStyles } from "react-jss";

import Letter from "./Letter";

const useStyles = createUseStyles({
  // TODO: Add shiver animation on non-word submit
  wordElement: {
    display: "flex"
  }
});

const Word = ({ word }) => {
  const { wordElement } = useStyles();
  return (
    <div className={wordElement}>
      {word.map((letterProps, index) => {
        return <Letter key={index.toString()} {...letterProps} />;
      })}
    </div>
  );
};

export default Word;
