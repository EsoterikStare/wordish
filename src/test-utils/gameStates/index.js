import loseGameStateJSON from './loseGameState.json';
import midGuessGameStateJSON from './midGuessGameState.json';
import newGameStateJSON from './newGameState.json';
import winGameStateJSON from './winGameState.json';

export const loseGameState = JSON.stringify(loseGameStateJSON);
export const midGuessGameState = JSON.stringify(midGuessGameStateJSON);
export const newGameState = JSON.stringify(newGameStateJSON);
export const winGameState = JSON.stringify(winGameStateJSON);
