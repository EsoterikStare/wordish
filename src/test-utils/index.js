export * from './gameStates';
export { default as setup } from './setup';
