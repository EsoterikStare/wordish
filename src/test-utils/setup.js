import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { node } from 'prop-types';
import React from 'react';

import { AppStateProvider } from '../AppStateContext';
import AppThemeProvider from '../AppThemeProvider';

const Wrapper = ({ children }) => (
  <AppStateProvider>
    <AppThemeProvider>{children}</AppThemeProvider>
  </AppStateProvider>
);
Wrapper.propTypes = {
  children: node.isRequired,
};
Wrapper.defaultProps = {};

const setup = jsx => {
  const { rerender, ...rest } = render(<Wrapper>{jsx}</Wrapper>);
  return {
    user: userEvent.setup(),
    rerender: newJsx => rerender(<Wrapper>{newJsx}</Wrapper>),
    ...rest,
  };
};

export default setup;
