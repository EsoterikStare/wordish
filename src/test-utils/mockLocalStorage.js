import { newGameState } from './gameStates';

const mockLocalStorage = (gameState = newGameState) => {
  const originalLocalStorage = { ...window.localStorage };
  const mockValues = {
    clear: jest.fn(),
    // Provide a known win state from localStorage mock that will be used by AppStateProvider.
    getItem: jest.fn(() => gameState),
    removeItem: jest.fn(),
    setItem: jest.fn(),
  };

  // Mock window.localStorage with assertable functions.
  Object.defineProperty(window, 'localStorage', {
    value: mockValues,
    writable: true,
  });

  const resetLocalStorage = () => {
    Object.defineProperty(window, 'localStorage', {
      value: originalLocalStorage,
    });
  };

  return {
    resetLocalStorage,
    ...mockValues,
  };
};

export default mockLocalStorage;
