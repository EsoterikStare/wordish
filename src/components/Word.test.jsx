import { screen } from '@testing-library/react';
import React from 'react';

import Word from './Word';

import { loseGameState, midGuessGameState, setup } from '../test-utils';

// Should have all three letter types, e.g. absent, present, and located.
const submittedWord = JSON.parse(loseGameState).previousGuesses[1];

// Should have some empty Letters as well as filled Letters.
const unsubmittedWord = JSON.parse(midGuessGameState).currentGuess;

describe('Word', () => {
  it('should be defined', () => {
    expect(Word).toBeDefined();
  });
  it('should render', () => {
    setup(<Word word={submittedWord} />);
    expect(screen.getByTestId('word-grid')).toBeInTheDocument();
  });
  it('should correctly arrange Letter components into a row', () => {
    // This also covers the 6-letter case.
    setup(<Word word={unsubmittedWord} />);
    expect(screen.getByTestId('word-grid')).toMatchSnapshot();
  });
});
