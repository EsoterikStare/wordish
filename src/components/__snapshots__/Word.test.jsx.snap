// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Word should correctly arrange Letter components into a row 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-1 css-1urxow7-MuiGrid-root"
  data-testid="word-grid"
>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    data-testid="letter-grid"
  >
    <div
      class="css-1i1qu0k"
    >
      S
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    data-testid="letter-grid"
  >
    <div
      class="css-1i1qu0k"
    >
      P
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
    data-testid="letter-grid"
  >
    <div
      class="css-1i1qu0k"
    >
      O
    </div>
  </div>
</div>
`;
